examples/mnist_acgan.py

```diff
@@ -185,6 +182,13 @@ def build_discriminator():
         num_batches = int(x_train.shape[0] / batch_size)
         progress_bar = Progbar(target=num_batches)
 
+        # don't train discriminator to produce class labels for generated
+        # images. To preserve total weight of the auxilary classifier,
+        # take real image samples with weight 2.
+        disc_sample_weight = [np.ones(2 * batch_size, dtype=np.float32),
```

Comment:
This uses `float32` for one but not the other; it's not really necessary for either (memory is not an issue and they get cast anyway). In any case, better to be consistent